### PR TITLE
planner, MPP: fix two bugs about group by duplicated and sort by exprs

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/3335,close https://github.com/pingcap/tics/issues/3337
close https://github.com/pingcap/tics/issues/3359

Problem Summary:
https://github.com/pingcap/tics/issues/3337 caused by 0 == 'gg'
https://github.com/pingcap/tics/issues/3335 caused by sort by items.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/testdata/integration_serial_suite_in.json [hunk 1]
```diff
@@ -369,7 +369,12 @@
       "desc format = 'brief' select /*+ hash_agg(),agg_to_cop() */ group_concat(col_0, id),count(distinct id),group_concat(col_1, id order by 1,2),min(col_0),avg(id) from ts",
       "desc format = 'brief' select /*+ hash_agg(),agg_to_cop() */ group_concat(distinct col_0, col_1, id),count(distinct id),group_concat(col_1, id order by 1,2),max(col_1),avg(id) from ts",
       "desc format = 'brief' select /*+ hash_agg(),agg_to_cop() */ group_concat(distinct col_0, col_1, id),count(distinct col_2),group_concat(col_1, id),max(col_1),avg(id) from ts",
-      "desc format = 'brief' select /*+ hash_agg(),agg_to_cop() */ group_concat(distinct col_0, col_1, id),count(distinct col_2),group_concat(col_1, id),max(col_1),avg(id) from ts group by col_0"
+      "desc format = 'brief' select /*+ hash_agg(),agg_to_cop() */ group_concat(distinct col_0, col_1, id),count(distinct col_2),group_concat(col_1, id),max(col_1),avg(id) from ts group by col_0",
```
Comment (windtalker) [inline planner/core/testdata/integration_serial_suite_in.json:372]: Add tests for group_concat(distinct same_cols/same_constants)

Comment (fzhedu) [reply planner/core/testdata/integration_serial_suite_in.json:372]: added

**Review by windtalker — APPROVED** [review]
LGTM

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**